RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,993 @@
+#include "RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexException.h"
+
+#include <cmath>
+#include <cassert>
+#include <limits>
+#include <iomanip>
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "vdt/vdtMath.h"
+
+using namespace std;
+
+namespace {
+  constexpr double epsilon = 1.0e-3;
+  constexpr double vertexSizeTime = 0.008;
+  constexpr double dtCutOff = 4.0;
+}
+
+DAClusterizerInZT_vect::DAClusterizerInZT_vect(const edm::ParameterSet& conf) {
+
+  // hardcoded parameters
+  maxIterations_ = 100;
+  mintrkweight_ = 0.5; // conf.getParameter<double>("mintrkweight");
+
+
+  // configurable debug outptut debug output
+  verbose_ = conf.getUntrackedParameter<bool> ("verbose", false);
+  zdumpcenter_ = conf.getUntrackedParameter<double> ("zdumpcenter", 0.);
+  zdumpwidth_ = conf.getUntrackedParameter<double> ("zdumpwidth", 20.);
+  
+  // configurable parameters
+  double Tmin = conf.getParameter<double> ("Tmin");
+  double Tpurge = conf.getParameter<double> ("Tpurge");
+  double Tstop = conf.getParameter<double> ("Tstop");
+  vertexSize_ = conf.getParameter<double> ("vertexSize");
+  coolingFactor_ = conf.getParameter<double> ("coolingFactor");
+  useTc_=true;
+  if(coolingFactor_<0){
+    coolingFactor_=-coolingFactor_; 
+    useTc_=false;
+  }
+  d0CutOff_ = conf.getParameter<double> ("d0CutOff");
+  dzCutOff_ = conf.getParameter<double> ("dzCutOff");
+  uniquetrkweight_ = conf.getParameter<double>("uniquetrkweight");
+  zmerge_ = conf.getParameter<double>("zmerge");
+  tmerge_ = conf.getParameter<double>("tmerge");
+
+  if(verbose_){
```

Comment:
cout statements need to be behind preprocessor flags or replaced with messagelogger (everywhere in this PR)